gpflow/base.py

```diff
@@ -53,6 +59,8 @@ def __init__(self,
         super().__init__()
 
         self._transform = transform
+        self.prior = prior
+        self.prior_on = prior_on
```

Comment:
```suggestion
        self.prior_on = PriorOn(prior_on)
```
together with the changes to the PriorOn class suggested above, this will allow a user to call
```python
Parameter(..., prior=..., prior_on="unconstrained")
```
instead of having to import the PriorOn enum class and the more cumbersome
```python
Parameter(..., prior=..., prior_on=PriorOn.UNCONSTRAINED)
```